Branch-SDK/src/main/java/io/branch/referral/BranchLinkGenerationException.kt

```diff
@@ -0,0 +1,61 @@
+package io.branch.referral
+
+/**
+ * Exception thrown during Branch link generation operations.
+ * 
+ * Specific exception types for different failure scenarios.
+ */
+sealed class BranchLinkGenerationException(
+    message: String,
+    cause: Throwable? = null
```

Comment:
Any reason why `Throwable` instead of `Exception`? Since `Throwable` is a parent of `Error`, I don't think we'd want to handle for example `OutOfMemoryError`, and the SDK only throws `Exception`